src/server/main_service.cc

```diff
@@ -2063,11 +2063,21 @@ GlobalState Service::GetGlobalState() const {
   return global_state_;
 }
 
-void Service::ConfigureHttpHandlers(util::HttpListenerBase* base) {
+void Service::ConfigureHttpHandlers(util::HttpListenerBase* base, bool is_admin) {
   // We set the password for the HTTP service unless it is only enabled on the
   // admin port and the admin port is password-less.
-  if (GetFlag(FLAGS_primary_port_http_enabled) || !GetFlag(FLAGS_admin_nopass)) {
-    base->SetPassword(GetPassword());
+  if (is_admin && GetFlag(FLAGS_admin_nopass)) {
+    base->SetAuthFunctor([](std::string_view path, std::string_view username,
+                            std::string_view password) { return username == "default"; });
```

Comment:
Why not always return `true` here?

Reply:
Because the `admin _nopass` flag is about the password token and not the user. Therefore, we still require a user with `username` default

Reply:
1. there is comment literally at line 2067 saying something else.
2. You can run 1.10 and see that we can actually access http page freely without any authorisation. 
3. Why do you think it is useful to anyone to have a dialog-box asking for a username and password where username is known and password is empty.

Reply:
I am asking - if you decide to change the existing product behaviour please ask me and be explicit about this.

Reply:
1. I quote:
```
  // We set the password for the HTTP service unless it is only enabled on the
  // admin port and the admin port is password-less.
``` 
Where in these two sentences it describes what happens with the `username`? It only talks about setting the password... Moreover, from the flags: 
```
    --admin_nopass (If set, would enable open admin access to console on the                             
      assigned port, without auth token needed.); default: false;
```

From my understanding the `Auth token` is the password not the username/password pair. Maybe we should be explicit here because personally, I did not understand the behavior that you described. And to be frank neither does DF handle this very well. See below. 

2. And you can run the following on 1.10:
   a. Execute this small test:
```
 20 async def test_no_password_on_admin(df_factory):
 22     with df_factory.create(
 23         port=1112,
 24         admin_port=1113,
 25         requirepass="XXX",
 26         primary_port_http_enabled=True,
 27         admin_nopass=None,
 28     ) as server:
 29         async with aiohttp.ClientSession() as session:
 30             resp = await session.get(f"http://localhost:{server.admin_port}/")
 31             assert resp.status == 200

```
it fails.
3. I got confused from the above bug.

p.s. I did not try to change the product but from the behavior I got I somehow thought this is sensible. I should have pinged. I now know better.
p.s.2 It's now fixed.

Reply:
The text around is flag was written when we had `--requirepass` so it makes sense it will only talk about the token. v1.10 does not require authentication if `--admin_nopass ` and that's the desired product behavior. Text indeed can be improved now. I would just replace `auth token` by `autharization`